<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>互花米草入侵的微生物AI模型构建 | 洋洋洒洒三两字</title><meta name="author" content="李金辉"><meta name="copyright" content="李金辉"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="ffffff"><meta name="description" content="研究方法搜集数据 文献 web网页检索  数据处理    FastQC检查    双端数据合并：join_paired_ended.py、EasyAmplicon    削减Phred至少20    read区域映射到全长16sRNA:Vsearch  基础数据库：Greengenes数据库13.8&#x2F;Unite7.2    聚类到OTU中    USEARCH格式的结果转换BIOM矩阵格">
<meta property="og:type" content="article">
<meta property="og:title" content="互花米草入侵的微生物AI模型构建">
<meta property="og:url" content="https://jinhuili-lab.github.io/blog/2022/01/05/ML-SA/index.html">
<meta property="og:site_name" content="洋洋洒洒三两字">
<meta property="og:description" content="研究方法搜集数据 文献 web网页检索  数据处理    FastQC检查    双端数据合并：join_paired_ended.py、EasyAmplicon    削减Phred至少20    read区域映射到全长16sRNA:Vsearch  基础数据库：Greengenes数据库13.8&#x2F;Unite7.2    聚类到OTU中    USEARCH格式的结果转换BIOM矩阵格">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg">
<meta property="article:published_time" content="2022-01-05T06:00:00.000Z">
<meta property="article:modified_time" content="2024-03-26T05:50:50.364Z">
<meta property="article:author" content="李金辉">
<meta property="article:tag" content="李金辉，博客">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg"><link rel="shortcut icon" href="https://jinhuili-lab.github.io/resume/assets/images/favicon.ico"><link rel="canonical" href="https://jinhuili-lab.github.io/blog/2022/01/05/ML-SA/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/blog/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/blog/',
  algolia: undefined,
  localSearch: undefined,
  translate: {"defaultEncoding":2,"translateDelay":0,"msgToTraditionalChinese":"繁","msgToSimplifiedChinese":"简"},
  noticeOutdate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js',
    buttonText: '加载更多'
  },
  isPhotoFigcaption: false,
  islazyload: true,
  isAnchor: true,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '互花米草入侵的微生物AI模型构建',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2024-03-26 00:50:50'
}</script><script>(win=>{
      win.saveToLocal = {
        set: (key, value, ttl) => {
          if (ttl === 0) return
          const now = Date.now()
          const expiry = now + ttl * 86400000
          const item = {
            value,
            expiry
          }
          localStorage.setItem(key, JSON.stringify(item))
        },
      
        get: key => {
          const itemStr = localStorage.getItem(key)
      
          if (!itemStr) {
            return undefined
          }
          const item = JSON.parse(itemStr)
          const now = Date.now()
      
          if (now > item.expiry) {
            localStorage.removeItem(key)
            return undefined
          }
          return item.value
        }
      }
    
      win.getScript = (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        script.onerror = reject
        script.onload = script.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          script.onload = script.onreadystatechange = null
          resolve()
        }

        Object.keys(attr).forEach(key => {
          script.setAttribute(key, attr[key])
        })

        document.head.appendChild(script)
      })
    
      win.getCSS = (url, id = false) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onerror = reject
        link.onload = link.onreadystatechange = function() {
          const loadState = this.readyState
          if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
          link.onload = link.onreadystatechange = null
          resolve()
        }
        document.head.appendChild(link)
      })
    
      win.activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', 'ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
        if (t === 'dark') activateDarkMode()
        else if (t === 'light') activateLightMode()
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
      const detectApple = () => {
        if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
          document.documentElement.classList.add('apple')
        }
      }
      detectApple()
    })(window)</script><meta name="generator" content="Hexo 7.2.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src= "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/blog/img/head.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/blog/archives/"><div class="headline">文章</div><div class="length-num">13</div></a><a href="/blog/tags/"><div class="headline">标签</div><div class="length-num">1</div></a><a href="/blog/categories/"><div class="headline">分类</div><div class="length-num">4</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/blog/../"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/blog/"><i class="fa-fw fas fa-book"></i><span> 博客</span></a></div><div class="menus_item"><a class="site-page" href="/blog/artitalk/"><i class="fa-fw fa-regular fa-comment-dots"></i><span> 说说</span></a></div><div class="menus_item"><a class="site-page" href="/blog/gallery/"><i class="fa-fw fas fa-photo-film"></i><span> 相册</span></a></div><div class="menus_item"><a class="site-page" href="/blog/../resume/"><i class="fa-fw fas fa-address-card"></i><span> 简历</span></a></div><div class="menus_item"><a class="site-page" href="/blog/../en/"><i class="fa-fw fa-fw fas fa-language"></i><span> 中/En</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="not-top-img fixed" id="page-header"><nav id="nav"><span id="blog-info"><a href="/blog/" title="洋洋洒洒三两字"><img class="site-icon" src= "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://jinhuili-lab.github.io/blog/img/head.png"/><span class="site-name">洋洋洒洒三两字</span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/blog/../"><i class="fa-fw fas fa-home"></i><span> 首页</span></a></div><div class="menus_item"><a class="site-page" href="/blog/"><i class="fa-fw fas fa-book"></i><span> 博客</span></a></div><div class="menus_item"><a class="site-page" href="/blog/artitalk/"><i class="fa-fw fa-regular fa-comment-dots"></i><span> 说说</span></a></div><div class="menus_item"><a class="site-page" href="/blog/gallery/"><i class="fa-fw fas fa-photo-film"></i><span> 相册</span></a></div><div class="menus_item"><a class="site-page" href="/blog/../resume/"><i class="fa-fw fas fa-address-card"></i><span> 简历</span></a></div><div class="menus_item"><a class="site-page" href="/blog/../en/"><i class="fa-fw fa-fw fas fa-language"></i><span> 中/En</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav></header><main class="layout" id="content-inner"><div id="post"><div id="post-info"><h1 class="post-title">互花米草入侵的微生物AI模型构建</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2022-01-05T06:00:00.000Z" title="发表于 2022-01-05 00:00:00">2022-01-05</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2024-03-26T05:50:50.364Z" title="更新于 2024-03-26 00:50:50">2024-03-26</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/blog/categories/%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%8E%E7%BE%A4%E8%90%BD%E5%BE%AE%E7%94%9F%E7%89%A9/">宏基因组与群落微生物</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="互花米草入侵的微生物AI模型构建"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div><article class="post-content" id="article-container"><h1 id="研究方法"><a href="#研究方法" class="headerlink" title="研究方法"></a>研究方法</h1><h2 id="搜集数据"><a href="#搜集数据" class="headerlink" title="搜集数据"></a>搜集数据</h2><ul>
<li>文献</li>
<li>web网页检索</li>
</ul>
<h2 id="数据处理"><a href="#数据处理" class="headerlink" title="数据处理"></a>数据处理</h2><ol>
<li>   FastQC检查</li>
<li>   双端数据合并：join_paired_ended.py、EasyAmplicon</li>
<li>   削减Phred至少20</li>
<li>   read区域映射到全长16sRNA:Vsearch  基础数据库：Greengenes数据库13.8&#x2F;Unite7.2</li>
<li>   聚类到OTU中</li>
<li>   USEARCH格式的结果转换BIOM矩阵格式（BIOM 2.1.5），biom</li>
<li>   分类群信息:summarize_taxa.py </li>
<li>   BIOM文件合并</li>
<li>   PCR偏好性：</li>
</ol>
<ul>
<li>两种过滤策略</li>
</ul>
<ul>
<li>(1)删除仅出现在所有样本中三分之一的otu;</li>
<li>(2)删除只出现在不到一半的独立数据集上的otu</li>
</ul>
<h2 id="机器学习"><a href="#机器学习" class="headerlink" title="机器学习"></a>机器学习</h2><h3 id="机器学习算法比较"><a href="#机器学习算法比较" class="headerlink" title="机器学习算法比较"></a>机器学习算法比较</h3><ul>
<li>三种机器学习方法：随机森林；支持向量机；逻辑回归</li>
<li>候选特征：OTU&#x2F;CSV</li>
<li>五重交叉验证建立分类器，进行性能评估</li>
<li>数据随机分成五份，5次迭代，80%的训练数据选择最小绝对收缩和选择算子作为特征选择方法</li>
<li>特征训练分类器(RF为1000棵树，SVM为径向核，其他参数默认设置</li>
<li>剩余20%验证数据</li>
<li>根据ROC曲线和曲线面积AUC评分。五次预测结果合并</li>
<li>交叉验证。AUC最高值，选取最佳分类器用于测试，独立验证预测能力</li>
<li>rfcv函数进行交叉验证选择合适的特征，使用varImpPlot函数来说明特惠总能在分类中的重要性，</li>
<li>R-ggplot对曲线可视化</li>
</ul>
<h3 id="机器学习建模"><a href="#机器学习建模" class="headerlink" title="机器学习建模"></a>机器学习建模</h3><ul>
<li>随机森林，鲁棒性，适用于微生物群落数据</li>
<li>RF包V4.6-14（R包）默认参数下对细菌相对丰度按界门纲目科属种以及<strong>OTU</strong>、<strong>ASV</strong>水平进行分类</li>
<li>对于每个分类单元，三分之二进行模型构建，三分之一用于预测。五次循环，评估准确性</li>
<li>24个研究758个样本的metadata训练集，生成两者分类模型，结果表明，准确率98.9%</li>
</ul>
<h3 id="两种样本之间的群落特征"><a href="#两种样本之间的群落特征" class="headerlink" title="两种样本之间的群落特征"></a>两种样本之间的群落特征</h3><ul>
<li>排除不同样本量对每个案例研究的影响，我们在样本总数大于80的情况下随机选取80样本</li>
<li>进行α多样性分析：利用alpha_diversity.py脚本计算QIIME中的Chao1、Shannon和Pielou均匀度指数</li>
<li>normalize_table.py对OTU剖面进行相对丰度标准化，用beta_diversity.py脚本编制β多样性计算和Bray-Curtis相似矩阵，基于Bray-Curtis相似矩阵，ggplot2生成主坐标分析PCoA图</li>
<li>SparCC python模块实现稀疏相关组成数据算法，得出OTU相对丰度之间的显著相关性，使用R包 igraph绘制共生网络。仅绘制R-corr绝对值&gt;0.4和P&lt;0.05的相关性，</li>
<li>描述网络拓扑结构，计算一组(即平均节点连通性、平均路径长度、直径、累积度分布、聚类系数和模块化)。R的vegan和igraph包</li>
<li>评估网络的非随机模式，使用igraph包将网络聚类稀疏、特征路径长度等结构属性与节点边相等的随机网络进行比较。</li>
</ul>
<h2 id="训练模型验证"><a href="#训练模型验证" class="headerlink" title="训练模型验证"></a>训练模型验证</h2><h3 id="土壤采集"><a href="#土壤采集" class="headerlink" title="土壤采集"></a>土壤采集</h3><ul>
<li>海南、广东、北京和江苏：香蕉、黄瓜、西瓜和百合</li>
<li><strong>文献划重点</strong>：跨地域，跨物种</li>
<li>0 ~ 20 cm土层随机取20个直径5 cm×20 cm的s型土芯，并组合为一个土壤样。新收集的土壤通过2毫米的网眼筛除植物碎片，并在−80°C保存，然后提取DNA。</li>
<li>试剂盒 ，样本ng，提取DNA，凝胶电泳和分光光度计评估DNA质量和数量。</li>
<li>引物：515F，292bp产物 PCR;Qubit®2.0荧光计(Thermo Scientific)和安捷伦Bioanalyzer 2100系统评估文库质量</li>
<li>Illumina Hiseq 2500平台测序，250 bp双端reads；Trimmomatic；测序原理</li>
</ul>
<h2 id="统计方法"><a href="#统计方法" class="headerlink" title="统计方法"></a>统计方法</h2><ul>
<li>从每个样本中随机抽取最小数量的序列来计算α多样性；alpha_diversity.py</li>
<li>非参数T检验用于确定两种样本中香农指数的差异</li>
<li>使用相对丰度来标准化OTU剖面，beta_diversity.py脚本编写Bray-Curtis相似矩阵</li>
<li>β多样性，</li>
<li>排列多元方差分析，用于确定beta多样性在两种处理之间是否存在差异</li>
<li>PCoA图由使用R package ggplot2创建的Bray-Curtis相似矩阵生成。</li>
</ul>
<h1 id="结果"><a href="#结果" class="headerlink" title="结果"></a>结果</h1><h2 id="多样性差异"><a href="#多样性差异" class="headerlink" title="多样性差异"></a>多样性差异</h2><ul>
<li>24个单独细菌研究，随机选择测序数据。19个个已发表，5个通过开放获取</li>
<li>合并得到的细菌OTU表包含了来自6个国家超过40000个分类群的758个单独土壤样本数据</li>
<li>为了防止单个研究中太多样本处于主导地位，对80个以上样本的研究进行随机过滤</li>
<li>将每个样本中的reads稀释到3000，计算Chao1.Shannon，Pielou均匀度指数。</li>
<li>结果表明，两种样本中α多样性没有明显差异</li>
<li>基于Bray-Curtis距离的主坐标分析PCoA显示，两种样本细菌群落差异显著(p&#x3D; 0.001,<br>PERMANOVA by Adonis)：患病土壤中厚壁菌门(Firmicutes)、拟杆菌门(Bacteroidetes)、Choroflexi和芽单胞菌门(Gemmatimonadetes)的相对丰度更高，而健康土壤中变形菌门(Proteobacteria)、放线菌门(Actinobacteria)和酸杆菌门(Acidobacteria)的相对丰度更高(图2c)。我们还发现，患病土壤(806)中的独特细菌OTUs比健康土壤(1</li>
</ul>
<h3 id="优化"><a href="#优化" class="headerlink" title="优化"></a>优化</h3><ul>
<li>为了突出微生物的共同特征，删除了所有样品中不到1&#x2F;3的或者在独立数据集中少于一半的OTU</li>
<li>过滤之后，表中1399个真菌OTU，用于下游分析，占细菌总数的37.7%</li>
<li>细菌群落的α多样性在两个样本中有明显差异</li>
</ul>
<h3 id="细菌和真菌群落都可以作为区分患病和健康土壤的生物标志物"><a href="#细菌和真菌群落都可以作为区分患病和健康土壤的生物标志物" class="headerlink" title="细菌和真菌群落都可以作为区分患病和健康土壤的生物标志物"></a>细菌和真菌群落都可以作为区分患病和健康土壤的生物标志物</h3><ul>
<li>通过随机森林、SVM和LR建立了三个机器学习模型。</li>
<li>模型导出的AUC和ROC曲线的考虑表明，射频模型显示出预测样品分类的最佳准确率</li>
<li>我们用细菌和真菌数据建立了在OTU水平上训练的模型显示出最高水平的分类准确性，对于两个模型分别为95%和92%。</li>
<li>为了验证细菌模型，从SRA数据库中13个独立数据集，包含26份A样本和321份正常样本。</li>
<li>细菌模型显示所有采集样本的平均准确率为93.4%，疾病样本为89。6%，健康样本95.4。</li>
</ul>
<h3 id="额外测试"><a href="#额外测试" class="headerlink" title="额外测试"></a>额外测试</h3><ul>
<li>采集了4种不同作物的田间土壤样本，20个</li>
<li>平均准确率为80%，疾病：90% 健康：70%</li>
<li>结果表明，模型可以合理预测特定土壤中是否发生病害</li>
</ul>
<h2 id="患病和健康土壤微生物特征识别"><a href="#患病和健康土壤微生物特征识别" class="headerlink" title="患病和健康土壤微生物特征识别"></a>患病和健康土壤微生物特征识别</h2><ul>
<li>进行了10倍的交叉验证，重复5次</li>
<li>评估OTUs在指标中潜在的重要性</li>
<li>在45个最相关的OTU上，交叉验证误差曲线趋于稳定，这些OTU被定义为相对丰度在0.16-4.84%之间的生物标志物类群</li>
<li>最重要的OTUGemmatimonadetes ，在病害样本中富集</li>
<li>在重要OTU中，33个在患病土壤中相对丰度高，12个在健康土壤中丰度高【FDR调整p&lt; 0.05, Wilcoxon秩和检验？】</li>
<li>简单介绍下种属</li>
</ul>
<h2 id="共现网络分析特征"><a href="#共现网络分析特征" class="headerlink" title="共现网络分析特征"></a>共现网络分析特征</h2><ul>
<li>用来评估特征间的相互作用，在本分析中只有显著相关性  (|R| &gt; 0.4, p&lt; 0.05)</li>
<li>健康微生物(细菌和真菌)特征网络中更多的节点和链接</li>
<li>健康土壤的平均程度、集中程度和聚集系数值高于患病土壤(补充表4)</li>
<li>一些OTUs，如那些被分类为杆菌科、原生菌科、慢生菌科、丛毛单胞菌科、链霉素科和红杆菌科(在健康特征网络中为红色)的OTUS，在健康细菌网络中包含更多的链接和更高的相对丰度</li>
<li>其他otu，如脂环酸杆菌科和鞘氨醇杆菌科(在患病特征网络中为红色)，在患病细菌网络中包含更多的链接和更高的相对丰度。</li>
<li>1111883和549433被分类为Gemmatimonadetes，通过模型识别显示出高重要性，与健康细菌网络(补充文件7、8)相比，在患病的细菌中包含更多的联系和更高的相对丰度</li>
</ul>
<h2 id="回顾"><a href="#回顾" class="headerlink" title="回顾"></a>回顾</h2><ul>
<li>将数据整合<ul>
<li>将序列比对到16s全长   ，参考数据库为greengene or Unite；已有同行这么干，在肠道领域</li>
<li>NOVO，从头方法，保留所有序列，包含更多OTU；但是对多个引物的元数据不够友好</li>
<li></li>
</ul>
</li>
<li>此外，在所有研究中，共有OTU只占15%，说明大部分都是特异性OTU：可能取决于取样偏好性、DNA提取偏倚和测序偏好性；RF算法已被广泛应用于微生物生态学研究，如模拟正常和高盐饮食小鼠的微生物区系差异</li>
<li>RF模型性能优秀，可以构建任何分类级别，但是我们发现OTU水平是最好结果。在其他的研究中，也有family水平的</li>
<li>案例研究证明，两种样本中确实有多样性的差异</li>
<li>土壤中真菌病原体数量的大量增加会破坏土壤真菌群落固有的平衡，导致多样性的减少</li>
<li>被分类为尖孢镰刀菌的OTU是区分患病土壤和健康土壤的最重要的真菌群落特征</li>
<li>虽然健康土壤微生物群落具体特征难以区分，但是患病土壤微生物主要特征是高数量的Foxysporum</li>
<li>虽然在患病土壤中尖孢镰刀菌的相对丰度很高，但研究表明其差异很大</li>
<li>健康土壤共生网络:我们还发现了一些相对丰度和位置较高的潜在有益细菌。其中包括链霉菌(micabili，图5)，已知链霉菌中有许多菌株能够抑制尖孢镰刀菌，促进植物生长;为区分健康土壤和病害土壤而建立的模型所产生的特征为与不同土壤类型的镰刀菌枯萎病作物相关的微生物群落的保守特征提供了令人信服的证据。<strong>我们找准碳四植物生长的特点来入手</strong></li>
<li>发病临界值为15%,表明植物发病率低于15%的土壤可以被认定为健康土壤</li>
<li>有趣的是，在训练数据集中不包含百合的基础上，百合的预测成功率非常高</li>
<li>未来期望对其他作物产生作用<strong>跨物种</strong></li>
<li>构建网站[](<a target="_blank" rel="noopener" href="https://wenta/">https://wenta</a> omicro.shinyapps.io&#x2F;foc_16s&#x2F;)序列数据中预测易感植物的枯萎病，并通过添加新的序列来改进模型。<br>模型是基于那些<strong>没有人为干扰</strong>的土壤的测序数据，如抗菌化合物的应用，可能会影响微生物群落的组成和模型的预测精度。杀菌剂处理土壤可能导致发病率降低<br>通过RF方法筛选出的45个细菌和40个真菌otu作为预测模型，可以认为是枯萎病相关核心微生物。镰刀菌和链霉菌种类的丰富度和多样性对土壤健康具有重要意义</li>
</ul>
<h1 id="数据"><a href="#数据" class="headerlink" title="数据"></a>数据</h1><ul>
<li>GSA数据库：CRA002340.</li>
<li>代码：  <a target="_blank" rel="noopener" href="https://github.com/taowenmicro/Wen-etal-200214-paper-code">https://github.com/taowenmicro/Wen-etal-200214-paper-code</a>.</li>
</ul>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>文章作者: </span><span class="post-copyright-info"><a href="https://jinhuili-lab.github.io/blog">李金辉</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>文章链接: </span><span class="post-copyright-info"><a href="https://jinhuili-lab.github.io/blog/2022/01/05/ML-SA/">https://jinhuili-lab.github.io/blog/2022/01/05/ML-SA/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="https://jinhuili-lab.github.io/blog" target="_blank">洋洋洒洒三两字</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="https://i.loli.net/2020/05/01/gkihqEjXxJ5UZ1C.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><div class="post-reward"><div class="reward-button"><i class="fas fa-qrcode"></i>赞助</div><div class="reward-main"><ul class="reward-all"><li class="reward-item"><a href="/blog/img/wechat.png" target="_blank"><img class="post-qr-code-img" src= "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/blog/img/wechat.png" alt="wechat"/></a><div class="post-qr-code-desc">wechat</div></li></ul></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/blog/2024/06/06/SturctureResearch/" title="The model family prediction"><img class="cover" src= "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://www.sigmaaldrich.com/deepweb/assets/sigmaaldrich/marketing/global/images/applications/protein-biology/protein-structure/protein-structure.jpg" onerror="onerror=null;src='/blog/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">The model family prediction</div></div></a></div><div class="next-post pull-right"><a href="/blog/2021/09/05/meta-8/" title="宏基因组出马，一个都别想跑"><img class="cover" src= "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://s2.loli.net/2024/03/22/OQfqPWVvEwMlbNa.jpg" onerror="onerror=null;src='/blog/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">宏基因组出马，一个都别想跑</div></div></a></div></nav><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="utterances-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content is-expand"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%A0%94%E7%A9%B6%E6%96%B9%E6%B3%95"><span class="toc-number">1.</span> <span class="toc-text">研究方法</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%90%9C%E9%9B%86%E6%95%B0%E6%8D%AE"><span class="toc-number">1.1.</span> <span class="toc-text">搜集数据</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86"><span class="toc-number">1.2.</span> <span class="toc-text">数据处理</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0"><span class="toc-number">1.3.</span> <span class="toc-text">机器学习</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0%E7%AE%97%E6%B3%95%E6%AF%94%E8%BE%83"><span class="toc-number">1.3.1.</span> <span class="toc-text">机器学习算法比较</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0%E5%BB%BA%E6%A8%A1"><span class="toc-number">1.3.2.</span> <span class="toc-text">机器学习建模</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%B8%A4%E7%A7%8D%E6%A0%B7%E6%9C%AC%E4%B9%8B%E9%97%B4%E7%9A%84%E7%BE%A4%E8%90%BD%E7%89%B9%E5%BE%81"><span class="toc-number">1.3.3.</span> <span class="toc-text">两种样本之间的群落特征</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%AE%AD%E7%BB%83%E6%A8%A1%E5%9E%8B%E9%AA%8C%E8%AF%81"><span class="toc-number">1.4.</span> <span class="toc-text">训练模型验证</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%9C%9F%E5%A3%A4%E9%87%87%E9%9B%86"><span class="toc-number">1.4.1.</span> <span class="toc-text">土壤采集</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%BB%9F%E8%AE%A1%E6%96%B9%E6%B3%95"><span class="toc-number">1.5.</span> <span class="toc-text">统计方法</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%93%E6%9E%9C"><span class="toc-number">2.</span> <span class="toc-text">结果</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%A4%9A%E6%A0%B7%E6%80%A7%E5%B7%AE%E5%BC%82"><span class="toc-number">2.1.</span> <span class="toc-text">多样性差异</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BC%98%E5%8C%96"><span class="toc-number">2.1.1.</span> <span class="toc-text">优化</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E7%BB%86%E8%8F%8C%E5%92%8C%E7%9C%9F%E8%8F%8C%E7%BE%A4%E8%90%BD%E9%83%BD%E5%8F%AF%E4%BB%A5%E4%BD%9C%E4%B8%BA%E5%8C%BA%E5%88%86%E6%82%A3%E7%97%85%E5%92%8C%E5%81%A5%E5%BA%B7%E5%9C%9F%E5%A3%A4%E7%9A%84%E7%94%9F%E7%89%A9%E6%A0%87%E5%BF%97%E7%89%A9"><span class="toc-number">2.1.2.</span> <span class="toc-text">细菌和真菌群落都可以作为区分患病和健康土壤的生物标志物</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E9%A2%9D%E5%A4%96%E6%B5%8B%E8%AF%95"><span class="toc-number">2.1.3.</span> <span class="toc-text">额外测试</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%82%A3%E7%97%85%E5%92%8C%E5%81%A5%E5%BA%B7%E5%9C%9F%E5%A3%A4%E5%BE%AE%E7%94%9F%E7%89%A9%E7%89%B9%E5%BE%81%E8%AF%86%E5%88%AB"><span class="toc-number">2.2.</span> <span class="toc-text">患病和健康土壤微生物特征识别</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%85%B1%E7%8E%B0%E7%BD%91%E7%BB%9C%E5%88%86%E6%9E%90%E7%89%B9%E5%BE%81"><span class="toc-number">2.3.</span> <span class="toc-text">共现网络分析特征</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%9B%9E%E9%A1%BE"><span class="toc-number">2.4.</span> <span class="toc-text">回顾</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE"><span class="toc-number">3.</span> <span class="toc-text">数据</span></a></li></ol></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2024 By 李金辉</div><div class="footer_custom_text">When you want something, all the universe conspires in helping you to achieve it.</div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="translateLink" type="button" title="简繁转换">繁</button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/blog/js/utils.js"></script><script src="/blog/js/main.js"></script><script src="/blog/js/tw_cn.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.umd.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page/instantpage.min.js" type="module"></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload/dist/lazyload.iife.min.js"></script><script>function panguFn () {
  if (typeof pangu === 'object') pangu.autoSpacingPage()
  else {
    getScript('https://cdn.jsdelivr.net/npm/pangu/dist/browser/pangu.min.js')
      .then(() => {
        pangu.autoSpacingPage()
      })
  }
}

function panguInit () {
  if (false){
    GLOBAL_CONFIG_SITE.isPost && panguFn()
  } else {
    panguFn()
  }
}

document.addEventListener('DOMContentLoaded', panguInit)</script><div class="js-pjax"><script>(() => {
  const loadUtterances = () => {
    let ele = document.createElement('script')
    ele.id = 'utterances_comment'
    ele.src = 'https://utteranc.es/client.js'
    ele.setAttribute('repo', 'jinhuili-lab/jinhuili-lab.github.io')
    ele.setAttribute('issue-term', 'og:title')
    const nowTheme = document.documentElement.getAttribute('data-theme') === 'dark' ? 'photon-dark' : 'github-light'
    ele.setAttribute('theme', nowTheme)
    ele.crossOrigin = 'anonymous'
    ele.async = true
    document.getElementById('utterances-wrap').appendChild(ele)
  }

  const utterancesTheme = theme => {
    const iframe = document.querySelector('.utterances-frame')
    if (iframe) {
      const theme = theme === 'dark' ? 'photon-dark' : 'github-light'
      const message = {
        type: 'set-theme',
        theme: theme
      };
      iframe.contentWindow.postMessage(message, 'https://utteranc.es');
    }
  }

  btf.addGlobalFn('themeChange', utterancesTheme, 'utterances')

  if ('Utterances' === 'Utterances' || !true) {
    if (true) btf.loadComment(document.getElementById('utterances-wrap'), loadUtterances)
    else loadUtterances()
  } else {
    window.loadOtherComment = loadUtterances
  }
})()</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/fireworks.min.js"></script><script defer="defer" id="ribbon" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-ribbon.min.js" size="150" alpha="0.6" zIndex="-1" mobile="false" data-click="false"></script><script defer="defer" id="fluttering_ribbon" mobile="true" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-fluttering-ribbon.min.js"></script><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-nest.min.js"></script><script id="click-heart" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/click-heart.min.js" async="async" mobile="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>